apps/livreGUI/transferFunctionEditor/RangeWidget.cpp

```diff
@@ -0,0 +1,182 @@
+﻿/* Copyright (c) 2017, EPFL/Blue Brain Project
+ *                     Daniel.Nachbaur@epfl.ch
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "RangeWidget.h"
+#include "Utils.h"
+
+#include <livreGUI/ui_RangeWidget.h>
+
+#include <QMouseEvent>
+#include <QQuickItem>
+#include <QQmlContext>
+
+#include <vmmlib/vector.hpp>
+
+namespace livre
+{
+
+RangeWidget::RangeWidget( QWidget* parent_ )
+    : QWidget( parent_ )
+    , _ui( new Ui::RangeWidget )
+{
+    _ui->setupUi( this );
+
+    _ui->background->installEventFilter( this );
+    _ui->background->rootContext()->setContextProperty( "model", this );
+
+    connect( _ui->fitToDataButton, &QPushButton::clicked, [&]
```

Comment:
although the result is the same, it would be enough for all the lambdas in this function to capture [this] instead of [&]